app.py

```diff
@@ -13,6 +13,10 @@
 def index():
     return render_template('index.html')
 
+@app.route('/grid_gen.html')
+def grip():
```

Comment:
I think you meant to call this `grid`?

```suggestion
def grid():
```